app/Repositories/SettingsRepository.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace App\Repositories;
+
+class SettingsRepository extends IniBaseRepository
+{
+    public function __construct()
+    {
+        return parent::__construct('settings');
+    }
+
+    public function getSetting($section, $subsection)
+    {
+        $settings = $this->all();
+
+        if ($settings == []) {
+
+            file_put_contents(prepare_file_path("{$this->dirName}/{$this->fileName}"), ini_encode($settings));
+
+            return $settings[$section][$subsection] = 'en';
```

Comment:
as we discussed earlier, please return the default settings structure

```suggestion
            $settings = $this->getDefaultSettings();

            file_put_contents(prepare_file_path("{$this->dirName}/{$this->fileName}"), ini_encode($settings));
```